src/name/name.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2021 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR
+// ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+// ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+// OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use super::DnsNameRef;
+
+/// A name that identifies a subject.
+#[derive(Clone, Copy)]
+#[non_exhaustive]
+pub enum Name<'a> {
```

Comment:
As this enum is opaque, I think the public API is missing some way for external users to build this. Possibly a `DnsNameRef::to_name()`?

Reply:
Probably just `impl<'a> From<DnsNameRef<'a> for Name<'a>`?

Reply:
The enum isn't opaque, but I've added the `From` implementation anyway.